static/js/tools/choropleth/choropleth.ts

```diff
@@ -0,0 +1,203 @@
+        geo.properties.value : "No Value"
+    document.getElementById("hover-text-display").innerHTML = name + " - " + geo_value
+    
+    // Highlight selected geo in black on hover.
+    d3.select(this).attr("class", "border-highlighted");
+  }
+  
+/**
+ * Clears output after leaving a geo.
+ */
+function mouseLeave() {
+  // Remove hover text.
+  document.getElementById("hover-text-display").innerHTML = "";
+
+  // Remove geo display effect.
+  d3.select(this).attr("class", "border");
+}
+
+/**
+ * Capture click event on geo and zooms
+ * user into that geo in the choropleth tool.
+ * @param {json} geo is the geoJson content for the clicked geo.
+ */
+function handleMapClick(geo) {
+  if (geo.properties.hasSublevel) {
+    redirectToGeo(geo.properties.geoId)
+  } else {
+    alert("This geo has no further sublevels!")
+  }
+}
+
+/**
+ * Builds a redirect link given the fields to include from search params.
+ * @param fields_to_include 
+ * @param from_api whether the url should be to the api or locally.
+ */
+function build_choropleth_url(fields_to_include, from_api) {
+  //TODO(iancostello): Make this path relative. 
+  var base_url = document.location.origin + document.location.pathname
+  if (from_api) {
+    base_url += "/download"
+  }
+  base_url += "?"
+  const urlParams = new URLSearchParams(window.location.search);
+  for (let index in fields_to_include) {
+    let arg_name = fields_to_include[index]
+    let arg_value = url.searchParams.get(arg_name);
+    if (arg_value != null) {
+      base_url += "&" + arg_name + "=" + arg_value
+    }
+  }
+  return base_url
+}
+
+/**
+ * Redirects the webclient to a particular geo. Handles breadcrumbs in redirect.
+ * @param {string} geoId to redirect to.
+ */
+function redirectToGeo(geoId) {
+  var base_url = build_choropleth_url(
+      ['statVar', 'perCapita', 'level', 'mdom'], false)
+  base_url += "&geoId=" + geoId
+  base_url += "&bc=" 
+  
+  // Add or create breadcrumbs field.
+  // TODO(iancostello): Use parent places api.
+  var breadcrumbs = url.searchParams.get("bc");
+  if (breadcrumbs != null && breadcrumbs != "") {
+    base_url += breadcrumbs + ";"
+  }
+  base_url += url.searchParams.get("geoId");
+  window.location.href = base_url
+}
+
+/**
+ * Generates the breadcrumbs text from browser url.
+ */
+function generateBreadCrumbs() {
+  var breadcrumbs = url.searchParams.get("bc");
+  if (breadcrumbs != null && breadcrumbs != "") {
+    var breadcrumbs_display = document.getElementById("breadcrumbs");
+    var crumbs = breadcrumbs.split(";")
+
+    // Build url for each reference in the breadcrumbs.
+    var base_url = build_choropleth_url(
+      ['statVar', 'perCapita', 'level', 'mdom'], false)
+    base_url += "&geoId=";
+
+    var breadcrumbs_upto = "";
+    for (let index in crumbs) {
+      let level_ref = crumbs[index];
+
+      if (level_ref != "") {
+        // TODO(iancostello): How should we sanitize this?
```

Comment:
We can turn this into a react component.

Reply:
Added